Services/JobRunner/Jobs/DataXDeployJob.cs

```diff
@@ -2,70 +2,70 @@
-
-        public DataXDeployJob(AppConfig config, ILogger logger)
-        {
-            _config = config;
-            _logger = logger;
-            
+namespace JobRunner.Jobs
+{
+    /// <summary>
+    /// Runs through a steel thread scenario for a DataX every 10 minutes to ensure DataX E2E works for saving and deploying a job.
+    /// </summary>
+    public class DataXDeployJob : IJob
+    {
+        private readonly ScenarioDescription _scenario;
+        private readonly AppConfig _config;
+        private readonly ILogger _logger;
+        private readonly int _scenarioCount = 1;
+
+        public DataXDeployJob(AppConfig config, ILogger logger)
+        {
+            _config = config;
+            _logger = logger;
+
             _scenario = new ScenarioDescription("DataXMainline",
-                DataXHost.AcquireToken,
-                DataXHost.SaveJob,
-                DataXHost.GenerateConfigs,
-                DataXHost.GetFlow,
-                DataXHost.RestartJob
-               );
-        }
+                DataXHost.AcquireToken,
+                DataXHost.SaveJob,
+                DataXHost.GenerateConfigs,
+                DataXHost.RestartJob,
+                DataXHost.GetFlow
+               );
+        }
         /// <summary>
         /// This is the method that gets called when the job starts running
         /// </summary>
-        /// <returns></returns>
-        public async Task RunAsync()
-        {
-            var server = _config.ServiceUrl;
-            var microsoftAuthority = _config.MicrosoftAuthority;
-            var dataHubIdentifier = _config.DataHubIdentifier;
-            if (string.IsNullOrWhiteSpace(server))
-            {
-                string errorMessage = "Server URL is not available.";
-                _logger.LogError(_scenario.Description, "JobRunner ScenarioTester", new Dictionary<string, string>() { { "scenario.errorMessage", errorMessage } });                
-
-                throw new InvalidOperationException(errorMessage);
-            }
-
-            using (var context = new ScenarioContext())
-            {
-                context[Context.ServiceUrl] = server;
-                context[Context.ApplicationId] = KeyVault.GetSecretFromKeyvault(_config.ServiceKeyVaultName, _config.ApplicationId);                
-                    
+        /// <returns></returns>
+        public async Task RunAsync()
+        {
+            var server = _config.ServiceUrl;
+            var microsoftAuthority = _config.MicrosoftAuthority;
+            var dataHubIdentifier = _config.DataHubIdentifier;
+            if (string.IsNullOrWhiteSpace(server))
+            {
+                string errorMessage = "Server URL is not available.";
+                _logger.LogError(_scenario.Description, "JobRunner ScenarioTester", new Dictionary<string, string>() { { "scenario.errorMessage", errorMessage } });
+
+                throw new InvalidOperationException(errorMessage);

```

Comment:
Looks like this is an ArgumentException?

Reply:
There is no argument being passed in. This is coming from the configuration settings file.